### PR TITLE
Fix kubectl-convert import known versions

### LINKED ISSUE #97729: kubectl-convert cannot convert Ingress objects
**What happened**:

Tried to convert a v1beta1 Ingress object to v1, and received the following error:

```
kubectl-convert -f v1beta1.yaml 
error: unable to decode "v1beta1.yaml": no kind "Ingress" is registered for version "networking.k8s.io/v1beta1" in scheme "k8s.io/kubernetes/pkg/api/legacyscheme/scheme.go:30"
```

**What you expected to happen**:

The conversion to work.

**How to reproduce it (as minimally and precisely as possible)**:

v1beta1.yaml:
```yaml
v1beta1.yaml 
apiVersion: networking.k8s.io/v1beta1
kind: Ingress
metadata:
  name: example
spec:
  backend:
    serviceName: default-backend
    servicePort: 80
  rules:
  - http:
      paths:
      - path: /testpath
        backend:
          serviceName: test
          servicePort: 80
```

Then run:
```sh
kubectl-convert -f v1beta1.yaml 
```

/sig cli
/assign @soltysh

### PR DESCRIPTION
**Pull request by wzshiming**
**What type of PR is this?**
/kind bug
/sig cli

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #97729

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
kubectl-convert: Fix a regression in 1.20 with `no kind "Ingress" is registered for version` error
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/kubectl/cmd/convert/import_known_versions.go [hunk 1]
```diff
@@ -19,6 +19,9 @@ package convert
 // These imports are the API groups the client will support.
 // TODO: Remove these manual install once we don't need legacy scheme in convert
 import (
+	_ "k8s.io/kubernetes/pkg/apis/admission/install"
+	_ "k8s.io/kubernetes/pkg/apis/admissionregistration/install"
+	_ "k8s.io/kubernetes/pkg/apis/apiserverinternal/install"
```
Comment (knight42) [inline pkg/kubectl/cmd/convert/import_known_versions.go:24]: Why do we need to import the internal APIs?

**Comment by liggitt** [comment]
Thanks for the PR. We need a unit test to ensure the API types defined and known to client-go continue to be registered here.

File: pkg/kubectl/cmd/convert/import_known_versions.go (hunk 1 above)
Comment (wzshiming) [reply pkg/kubectl/cmd/convert/import_known_versions.go:24]: You are right, there is no need to import the internal.

File: pkg/kubectl/cmd/convert/convert_test.go [hunk 2]
```diff
@@ -90,6 +90,16 @@ func TestConvertObject(t *testing.T) {
 				},
 			},
 		},
+		{
```
Comment (liggitt) [inline pkg/kubectl/cmd/convert/convert_test.go:93]: this test is fine for ingress specifically but will not catch other missing types.

add a test like this in this file:
```go

func TestKnownVersions(t *testing.T) {
	legacytypes := legacyscheme.Scheme.AllKnownTypes()
	alreadyErroredGroups := map[string]bool{}
	var gvks []schema.GroupVersionKind
	for gvk := range scheme.Scheme.AllKnownTypes() {
		gvks = append(gvks, gvk)
	}
	sort.Slice(gvks, func(i, j int) bool {
		if isWatchEvent1, isWatchEvent2 := gvks[i].Kind == "WatchEvent", gvks[j].Kind == "WatchEvent"; isWatchEvent1 != isWatchEvent2 {
			return isWatchEvent2
		}
		if isList1, isList2 := strings.HasSuffix(gvks[i].Kind, "List"), strings.HasSuffix(gvks[j].Kind, "List"); isList1 != isList2 {
			return isList2
		}
		if isOptions1, isOptions2 := strings.HasSuffix(gvks[i].Kind, "Options"), strings.HasSuffix(gvks[j].Kind, "Options"); isOptions1 != isOptions2 {
			return isOptions2
		}
		if isInternal1, isInternal2 := gvks[i].Group == runtime.APIVersionInternal, gvks[j].Group == runtime.APIVersionInternal; isInternal1 != isInternal2 {
			return isInternal2
		}
		return gvks[i].String() < gvks[j].String()
	})
	for _, gvk := range gvks {
		if alreadyErroredGroups[gvk.Group] {
			continue
		}
		if _, legacyregistered := legacytypes[gvk]; !legacyregistered {
			t.Errorf("%v is not registered in legacyscheme. Add group %q (all internal and external versions) to convert/import_known_versions.go", gvk, gvk.Group)
			alreadyErroredGroups[gvk.Group] = true
		}
	}
}
```

**Comment by liggitt** [comment]
thanks, once this merges, can you backport it to 1.20?

**Pull request merged**